test/unit/ngsiv2/examples/contextRequests/updateContextCompressTimestamp1.json

```diff
@@ -0,0 +1,10 @@
+{
+  "state": {
+    "type": "Boolean",
+    "value": "true"
+  },
+  "The Target Value": {
+    "type": "DateTime",
+    "value": "+002007-11-03T13:18:05"
```

Comment:
Not sure whtat "+00" means. I'm used to ISO8601 dates such "2007-11-03T13:18:05" but +00 is unknwon to me... Maybe it represents timezone?

Reply:
I am not an expert in ISO8601 and in fact this example is just an adaptation of an existing one for NGSIv1(https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/examples/contextRequests/updateContextCompressTimestamp1.json). Nevertheless, it seems that this kind of format is included in ISO8601 standard for expanded representation:
* In https://en.wikipedia.org/wiki/ISO_8601:

>  An expanded year representation [±YYYYY] must have an agreed-upon number of extra year digits beyond the four-digit minimum, and it must be prefixed with a + or − sign[19] instead of the more common AD/BC (or CE/BCE) notation.

* In https://es.wikipedia.org/wiki/ISO_8601, you can see some examples.

I understand that this is the functionality implemented in this plugin. From https://github.com/telefonicaid/iotagent-node-lib#timestamp-compression-plugin-compresstimestamp:

> This plugins change all the timestamp attributes found in the entity, and all the timestamp metadata found in any attribute, from the basic complete calendar timestamp of the ISO8601 (e.g.: 20071103T131805) to the extended complete calendar timestamp (e.g.: +002007-11-03T13:18)

Reply:
Thank you for your detailed explanation :)

Not sure how much is this plugin used in the main IoTAgents using the library. That information could help to "calibrate" the care we should put on this plugin migration.

In this sense, could you have a look to the following IOTAs to check which one is using this plugin, please? (I guess it is a matter of looking in IOTAs code for the handler exposed by the plugin, so it should be easy, if it is more complex, please tell me so)

* https://github.com/telefonicaid/iotagent-ul
* https://github.com/telefonicaid/iotagent-json
* https://github.com/telefonicaid/lightweightm2m-iotagent

Thanks!

Reply:
It is used iotagent-json. In  [iotagent-json.js](https://github.com/telefonicaid/iotagent-json/blob/master/lib/iotagent-json.js):

>             if (config.getConfig().iota && config.getConfig().iota.compressTimestamp) {
                iotAgentLib.addUpdateMiddleware(iotAgentLib.dataPlugins.compressTimestamp.update);
                iotAgentLib.addQueryMiddleware(iotAgentLib.dataPlugins.compressTimestamp.query);
            }

Reply:
Just to to check if I understanding correctly...

* iotagent-json:
   * It uses Time Comrpession plugin
   * It doesn't uses any other plugin in https://github.com/telefonicaid/iotagent-node-lib#provided-plugins
* iotagent-ul
   * It doesn't uses Time Compression plugin
   * It uses all the others plugins in https://github.com/telefonicaid/iotagent-node-lib#provided-plugins
* lightweightm2m-iotagent
   * It doesn't uses any plugin in https://github.com/telefonicaid/iotagent-node-lib#provided-plugins at all

Is that correct?

Reply:
Just a minor correction:

* iotagent-json:
   * It uses all the plugins in https://github.com/telefonicaid/iotagent-node-lib#provided-plugins

Reply:
Ok. Thanks for the analysis. NTC